### PR TITLE
Support Apple Silicon

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Signed-off-by: Xiaoguang Sun <sunxiaoguang@zhihu.com>

### What problem does this PR solve? 
Make tiup build natively on Apple Silicon M1

### What is changed and how it works?
Upgrade golangci-lint to versions that support M1 and fix lints with newer version. Also changed a little bit to Makefile in order to build using native arch as default

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

- Minor lint changes
- Makefile changes

Side effects

- NA

Related changes

 - Need to update the documentation

Release notes:

```release-note
Support Apple Silicon M1
```

### CODE REVIEW

File: tools/check/check-lint.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
 # check for io/ioutil
-GREP_IOUTIL=`find -name "*.go" | xargs grep -ns "io\/ioutil"`
+GREP_IOUTIL=`find --name "*.go" | xargs grep -ns "io\/ioutil"`
```
Comment (AstroProfundis) [inline tools/check/check-lint.sh:4]: GNU find uses `-name` as the param, `--name` errors with `find: unknown predicate '--name'`. Is `find` in OSX using `--name`?

Comment (sunxiaoguang) [reply tools/check/check-lint.sh:4]: Sorry, it was a mistake which weren't suppose to be committed. Let me revert this file.

**Comment by AstroProfundis** [comment]
This is part of #1122

**Comment by sunxiaoguang** [comment]
> This is part of #1122

Oops, didn't notice that, feel free to close this one if it's already done.

**Comment by AstroProfundis** [comment]
> > This is part of #1122
> 
> Oops, didn't notice that, feel free to close this one if it's already done.

That's totally fine, there is no conflicts of other PRs related to that issue.

**Review by AstroProfundis — APPROVED** [review]

**Pull request merged**